src/textual/widgets/_input.py

```diff
@@ -168,11 +176,18 @@ class Input(Widget, can_focus=True):
     width = reactive(1)
     _cursor_visible = reactive(True)
     password = reactive(False)
-    max_size: reactive[int | None] = reactive(None)
     suggester: Suggester | None
     """The suggester used to provide completions as the user types."""
     _suggestion = reactive("")
     """A completion suggestion for the current value in the input."""
+    restrict = var["str | None"](None)
+    """A regular expression that must match incoming characters."""
+    type = var("text")
+    """The type of the input."""
+    max_length = var["int | None"](None)
+    """The maximum length of the input, in characters."""
+    valid_empty = var(False)
```

Comment:
I think there's an issue in that after mounting (i.e. at runtime), if the `value` of the input is empty, and the user changes `valid_empty` to `True`, the CSS classes will be incorrect (it'll probably remain `-invalid` instead of being updated immediately to `-valid`).